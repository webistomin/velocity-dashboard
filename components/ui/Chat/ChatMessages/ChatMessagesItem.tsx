import Vue, { CreateElement, RenderContext, VNode, PropType } from 'vue';

import { BaseThumbnail } from 'components/base/BaseThumbnail/BaseThumbnail';
import { IChatMessage } from 'common/types/chat/chat-list';

export interface IChatMessagesItemProps {
  index: number;
  source: object;
}

export const ChatMessagesItem = Vue.extend({
  functional: true,
  props: {
    index: {
      type: Number,
      default: 0,
    },
    source: {
      type: Object,
      default: {} as PropType<IChatMessage>,
    },
  },
  render(_h: CreateElement, ctx: RenderContext<IChatMessagesItemProps>): VNode {
    const { staticClass, class: cls } = ctx.data;
    const { sender, messages } = ctx.props.source as IChatMessage;
    return (
      <div class={`chat-messages__item chat-messages__item_${sender.type} list-item ${staticClass || ''} ${cls || ''}`}>
        <BaseThumbnail class='chat-messages__avatar' image={sender.avatar} size='s' alt={sender.name} />
        <div class='chat-messages__holder'>
          <strong class='chat-messages__name'>{sender.name}</strong>
          <ul class='chat-messages__group-messages list'>
            {messages.map((message) => {
              return (
                <li class='chat-messages__group-item list-item'>
                  <p class='chat-messages__message paragraph'>{message.message}</p>
                  <time class='chat-messages__date'>{message.time}</time>
                </li>
              );
            })}
          </ul>
        </div>
      </div>
    );
  },
});
