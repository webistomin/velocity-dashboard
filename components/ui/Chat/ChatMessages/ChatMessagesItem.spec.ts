import { shallowMount } from '@vue/test-utils';
import { ChatMessagesItem } from './ChatMessagesItem';

describe('ChatMessagesItem', () => {
  const factory = (propsData = {}, options = {}) => {
    return shallowMount(ChatMessagesItem, {
      propsData: {
        source: {
          id: 'message-1',
          sender: {
            type: 'other',
            avatar: '/img/avatar.png',
            name: 'Alexey Istomin',
          },
          messages: [
            {
              message: 'Hi, I have trouble with trip #12422',
              time: '10:00',
            },
          ],
        },
        index: 0,
        ...propsData,
      },
      stubs: ['v-lazy-image'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });
});
