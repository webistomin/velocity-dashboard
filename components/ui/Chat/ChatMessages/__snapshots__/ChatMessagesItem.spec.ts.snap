// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatMessagesItem Matches snapshot 1`] = `
"<div class=\\"chat-messages__item chat-messages__item_other list-item  \\"><span class=\\"base-thumbnail base-thumbnail_size_s chat-messages__avatar  \\"><picture class=\\"base-thumbnail__picture picture\\"><v-lazy-image-stub src=\\"/img/avatar.png\\" alt=\\"Alexey Istomin\\" data-jest=\\"base-thumbnail__img\\" class=\\"base-thumbnail__img image\\"></v-lazy-image-stub></picture></span>
  <div class=\\"chat-messages__holder\\"><strong class=\\"chat-messages__name\\">Alexey Istomin</strong>
    <ul class=\\"chat-messages__group-messages list\\">
      <li class=\\"chat-messages__group-item list-item\\">
        <p class=\\"chat-messages__message paragraph\\">Hi, I have trouble with trip #12422</p><time class=\\"chat-messages__date\\">10:00</time>
      </li>
    </ul>
  </div>
</div>"
`;
